### PR TITLE
fix compiler error on macos 13.3

### PR DESCRIPTION
**Pull request by wmliang**
there is some compile error on the new macos 13.3 with Apple clang version 14.0.3 (clang-1403.0.22.14.1)

### CODE REVIEW

**Comment by wmliang** [comment]
error message when compiling hfuzz_cc/hfuzz-cc.c
["clang: error: '-x objective-c' after last input file has no effect"](clang: error: '-x objective-c' after last input file has no effect [-Werror,-Wunused-command-line-argument])

File: mac/arch.c [hunk 1]
```diff
@@ -403,7 +403,7 @@ void arch_reapChild(run_t* run) {
 void arch_reapKill(void) {
 }
 
-void* wait_for_exception() {
+void* wait_for_exception(void) {
```
Comment (devnexen) [inline mac/arch.c:406]: if you look at the signature you provided yourself below, it s wrong here.

File: mac/arch.c [hunk 2]
```diff
@@ -461,7 +461,7 @@ bool arch_archInit(honggfuzz_t* hfuzz) {
      */
     pthread_t exception_thread;
 
-    if (pthread_create(&exception_thread, NULL, wait_for_exception, 0)) {
+    if (pthread_create(&exception_thread, NULL, (void * _Nullable (* _Nonnull)(void * _Nullable))wait_for_exception, 0)) {
```
Comment (devnexen) [inline mac/arch.c:464]: I think you re "pushing" a little here :-) not sure the cast is so necessary once you fix above.

**Comment by robertswiecki** [comment]
Please re-open if you decide to sign the CLA.

**Pull request closed without merging**